samples/secret-santa/secret-santa.ipynb

```diff
@@ -0,0 +1,276 @@
+    "\r\n",
+    "```python\r\n",
+    "terms = [Term(c = 13.0, indices = []), Term(c=17.0, indices = [0]) , Term(c = 23.0, indices = 1, 3, 77)] \r\n",
+    "```\r\n",
+    "\r\n",
+    "> **Note:** See [this documentation page](https://docs.microsoft.com/azure/quantum/quickstart-microsoft-qio?pivots=platform-microsoft#express-a-simple-problem) for further information."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Binary formulation for the Secret Santa problem\r\n",
+    "\r\n",
+    "To represent the Secret Santa problem, we can use six binary variables, as outlined in the Scenario Table below:\r\n",
+    "\r\n",
+    "|- buys ->|**Vincent**|**Tess**|**Uma**|\r\n",
+    "|--|--|--|--|\r\n",
+    "|**Vincent**|--|$x_0$|$x_1$|\r\n",
+    "|**Tess**|$x_2$|--|$x_3$|\r\n",
+    "|**Uma**|$x_3$|$x_4$|--|\r\n",
+    "\r\n",
+    "The constraints for the problem can be expressed as doing logical ANDs ($ \\land $) of variables that are EXLUSIVE-ORd ($ \\oplus $) together, like this:\r\n",
+    "\r\n",
+    "$\r\n",
+    "( x_0 \\oplus x_1 ) \\land ( x_2 \\oplus x_3 ) \\land ( x_4 \\oplus x_5 ) \\land ( x_2 \\oplus x_4 ) \\land ( x_0 \\oplus x_5 ) \\land ( x_1 \\oplus x_3 )\r\n",
+    "$\r\n",
+    "\r\n",
+    "$\r\n",
+    "\\text{ where } x_i \\in \\{ 0,1 \\} \r\n",
+    "$"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "\r\n",
+    "The truth table for exclusive or ($ \\oplus $) is shown below (one variable or the other is **one**, but not both):\r\n",
+    "\r\n",
+    "|$x_0$|$x_1$|$x_0 \\oplus x_1$|\r\n",
+    "|--|--|--|\r\n",
+    "|0|0|0|\r\n",
+    "|0|1|1|\r\n",
+    "|1|0|1|\r\n",
+    "|1|1|0|\r\n",
+    "\r\n",
+    "Using this truth table, we can see how the constraints are derived. Looking at the Scenario Table defined previously:\r\n",
+    "\r\n",
+    "- Reading the first **row** of the table, Vincent may buy a gift and write a poem for Tess or for Uma, but not both.\r\n",
+    "- Reading the first **column** of the table, Vincent may receive a gift and poem from Tess or from Uma, but not both.\r\n",
+    "\r\n",
+    "More generally:\r\n",
+    "\r\n",
+    "- Each person should give and receive **exactly one** gift from one other person in the group.\r\n",
+    "  - If a person gives more or less than one gift, or receives more or less than one gift, this constraint has been violated and the solution will not be valid."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "So, how do we represent $ ( x_0 \\oplus x_1 ) $ in a binary format that Azure Quantum will understand?\r\n",
+    "\r\n",
+    "Keeping in mind that we want to **minimize** our cost function, let's try to use the following representation:\r\n",
+    "\r\n",
+    "$ ( x_0 + x_1 - 1 )^2 $  \r\n",
+    "\r\n",
+    "Let's check the truth table for this formulation:\r\n",
+    "\r\n",
+    "|$x_0$|$x_1$|$(x_0 + x_1 - 1)^2$|\r\n",
```

Comment:
A suggestion: like in the shipping sample, we have a follow-up section on how to simplify the problem formulation, I think we could have a similar additional section which could show how to phrase this same problem using `SLCTerms`. I don't suggest doing it in this PR, just mostly making a note...

Reply:
Just got to the note at the end that suggests this!